src/xopen/__init__.py

```diff
@@ -93,6 +95,29 @@ def _set_pipe_size_to_max(fd: int):
     fcntl.fcntl(fd, fcntl.F_SETPIPE_SZ, _MAX_PIPE_SIZE)
 
 
+def _can_read_concatenated_gz(program: str) -> bool:
+    """
+    Check if a concatenated gzip file can be read properly. Not all deflate
+    programs handle this properly.
+    """
+    fd, temp_path = tempfile.mkstemp(suffix=".gz", prefix="xopen.")
```

Comment:
You could use `tempfile.NamedTemporaryFile` to simplify this a bit.

Reply:
That would delete the file upon closing. And I want to close the file before I read it with igzip. Windows does not handle it well if two process open the same file (according to the Python documentation).

Reply:
Ah yes, now I remember why I tend to use `TemporaryDirectory` most of the time ...